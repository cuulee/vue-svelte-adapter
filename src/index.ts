import Vue, { VueConstructor } from 'vue'

export function toVue(Ctor: any): VueConstructor {
  return Vue.extend({
    mounted() {
      ;(this as any).instance = new Ctor({
        target: this.$el
      })
    },

    beforeDestroy() {
      const { instance } = this as any
      instance.destroy()
    },

    render(h) {
      return h('div')
    }
  })
}
