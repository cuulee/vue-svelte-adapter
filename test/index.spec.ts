import { mount } from '@vue/test-utils'
import { toVue } from '../src/index'
import Basic from './fixtures/Basic.html'

describe('Vue Svelte Adapter', () => {
  it('renders template', () => {
    const BasicVue = toVue(Basic)
    const wrapper = mount(BasicVue)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
