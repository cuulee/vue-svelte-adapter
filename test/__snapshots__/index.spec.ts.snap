// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vue Svelte Adapter renders template 1`] = `"<div><h1>Hello</h1></div>"`;
